apps/dashboard/src/main.tsx

```diff
@@ -1,27 +1,55 @@
-import { StrictMode } from "react";
-import { createRoot } from "react-dom/client";
-import { createBrowserRouter, RouterProvider } from "react-router-dom";
-import "./index.css";
-import ErrorPage from "@/components/error-page";
-import Root from "@/routes/root";
-import { Workflows } from "@/routes/workflows";
+import { StrictMode } from 'react';
+import { createBrowserRouter, RouterProvider } from 'react-router-dom';
+import { createRoot } from 'react-dom/client';
+import ErrorPage from '@/components/error-page';
+import { Root, AuthLayout, DashboardLayout } from './layout';
+import { Workflows, SignIn, SignUp, OrganizationList } from '@/pages';
+import { ProtectedRoutes } from './components/protected-routes';
+import './index.css';
 
 const router = createBrowserRouter([
   {
-    path: "/",
     element: <Root />,
     errorElement: <ErrorPage />,
     children: [
       {
-        path: "workflows",
-        element: <Workflows />,
+        path: 'auth',
+        element: <AuthLayout />,
+        children: [
+          {
+            path: 'sign-in/*',
+            element: <SignIn />,
+          },
+          {
+            path: 'sign-up/*',
+            element: <SignUp />,
+          },
+          {
+            path: 'sign-up/organization-list',
+            element: <OrganizationList />,
+          },
+        ],
+      },
+      {
+        path: '/',
+        element: (
+          <ProtectedRoutes>
+            <DashboardLayout />
+          </ProtectedRoutes>
+        ),
+        children: [
+          {
+            path: 'workflows',
+            element: <Workflows />,
+          },
+        ],
```

Comment:
The protected dashboard route `/workflows`.

The important note: because of the way the react-router works and picks the routes for rendering, the protected routes `/*` should be defined at the end of `Root.children` array. 
First, it checks and matches with the public routes. Otherwise, it checks against the protected routes.